src/net.cpp

```diff
@@ -1966,7 +1964,9 @@ bool CConnman::OpenNetworkConnection(const CAddress& addrConnect, bool fCountFai
     if (fAddnode)
         pnode->fAddnode = true;
 
-    GetNodeSignals().InitializeNode(pnode, *this);
+    if (messageInterface) {
+        messageInterface->InitializeNode(pnode);
```

Comment:
Heh, I went back and forth on this. Was waiting for someone to complain. Agreed.